lib/service.js

```diff
@@ -3091,7 +3104,99 @@
             });
             
             return req;
-        }
+        },
+        
+        doesFileExist: async function(filename) {
+            try {
+                var that = this;
+
+                // 1. Fetch files list
+                var response = await this.get("",  {__conf: filename}, null, true);
+
+                // 2. Filter the files
+                var files = response 
+                                && response.body 
+                                && response.body.entry 
+                                && response.body.entry.filter(f => f.name === filename);
+                
+                // 3. Check if the file exists
+                if(files && files.length == 0) {
+                    return { code:"FILE_NOT_FOUND", message: `File with the name '${filename}' does not exist.`};
+                }
+
+                // 4. Create a local instance
+                var configurationFile = new root.ConfigurationFile(that.service, filename);
+
+                // 5. Load the file content
+                var fetchedFile = await configurationFile.fetchAsync();
+
+                return {file: fetchedFile};
+            }
+            catch(error) {
+                return { errorCode:"SYSTEM_ERROR", errorMessage: error};
```

Comment:
I was thinking for consistency with the rest of the SDK that probably re-throwing the error would make sense in this case - and throwing a new error on line 3124 above, did you try that approach? I'm concerned that creating new fields `code` and `errorCode` for this particular implementation are not going to be intuitive / introducing new fields that aren't used elsewhere

Reply:
Create a new file if it doesn't exist instead of throwing an error.
Throw an error if the stanza doesn't exist. right?

Reply:
So actually thinking about it again I think we want to rename the functions a bit and change the return format a bit in an effort to simplify. Can we change `doesFileExist` -> `getConfFile` and `doesStanzaExist` -> `getStanza`. This is because we are aren't actually simply returning a bool the way the original names imply. Also - rather than returning objects with either a file/stanza property or errorCode+message, let's return either the name of the file or stanza (if found) or `null` (if not found). And for this particular line of code where another error has occurred (not the case where the file is not found) let's allow that error to propagate, so I think removing the try/catch block makes sense.

Reply:
- [ ] rename functions to getConfFile, getStanza
- [ ] change output from `{file: myfile}` or {errorMessage:...} to return myfile or `null`
- [ ] remove try/catch and allow errors to propagate (if not related to file or stanza not existing

Reply:
Done